### PR TITLE
Fixes #22862 - add async_ssh param

### PR DESCRIPTION
**Pull request by chris1984**
Test failures are unrelated, on master it was still failing

### CODE REVIEW

File: templates/plugin/dynflow_core.yml.erb [hunk 1]
```diff
@@ -23,6 +23,13 @@
 :foreman_ssl_key: <%= scope.lookupvar("foreman_proxy::foreman_ssl_key") %>
 <% end -%>
 
+# make all remote execution jobs run asynchronously.
+<% if [nil, :undefined, :undef].include?(scope.lookupvar("foreman_proxy::plugin::dynflow::async_ssh")) -%>
+:async_ssh: false
```
Comment (ekohl) [inline templates/plugin/dynflow_core.yml.erb]: If you treat ```undef``` as ```false``` then I'd do that in ```params.pp``` and make it non-optional.

Comment (chris1984) [reply templates/plugin/dynflow_core.yml.erb]: Thanks, @ekohl updated to non-optional

File: manifests/plugin/dynflow.pp [hunk 2]
```diff
@@ -24,6 +24,8 @@
 #
 # $open_file_limit::       Limit number of open files - Only Red Hat Operating Systems with Software Collections.
 #
+# $async_ssh::             Make all remote execution jobs run asynchronously. True/False
```
Comment (ekohl) [inline manifests/plugin/dynflow.pp]: The type boolean implies it's true/false.

File: templates/plugin/dynflow_core.yml.erb [hunk 3]
```diff
@@ -23,6 +23,13 @@
 :foreman_ssl_key: <%= scope.lookupvar("foreman_proxy::foreman_ssl_key") %>
 <% end -%>
 
+# make all remote execution jobs run asynchronously.
+<% if [nil, :undefined, :undef].include?(scope.lookupvar("foreman_proxy::plugin::dynflow::async_ssh")) -%>
```
Comment (ekohl) [inline templates/plugin/dynflow_core.yml.erb]: You can remove this if block now it's mandatory.

**Comment by chris1984** [comment]
@ekohl updated.

File: manifests/plugin/dynflow.pp [hunk 4]
```diff
@@ -24,6 +24,8 @@
 #
 # $open_file_limit::       Limit number of open files - Only Red Hat Operating Systems with Software Collections.
 #
+# $async_ssh::             Make all remote execution jobs run asynchronously.
```
Comment (ekohl) [inline manifests/plugin/dynflow.pp]: Would here and in the config file it be better to write ```Whether to run remote execution jobs asynchronously```?

**Comment by chris1984** [comment]
@ekohl updated

**Comment by ekohl** [comment]
@chris1984 you should move the parameter to the correct class (https://github.com/theforeman/puppet-foreman_proxy/blob/master/manifests/plugin/remote_execution/ssh.pp) rather than just the correct template.

**Comment by chris1984** [comment]
@ekohl updated hopefully last time.

**Comment by evgeni** [comment]
LGTM, I'll leave it to @ekohl for final review though

**Comment by ekohl** [comment]
I opened https://github.com/theforeman/puppet-foreman_proxy/pull/420 to see if it fixes the tests.

**Comment by mmoll** [comment]
@chris1984 please rebase :)

**Comment by chris1984** [comment]
@mmoll rebased :)

**Comment by mmoll** [comment]
merged, thanks @chris1984!

**Pull request merged**